eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Deleg.hs

```diff
@@ -312,37 +319,26 @@ processDelegationInternal preserveIncorrectDelegation stakeCred mCurDelegatee ne
       cState
         & certDStateL . accountsL
           %~ adjustAccountState (stakePoolDelegationAccountStateL ?~ stakePool) stakeCred
+        & maybe
+          (certPStateL . psStakePoolsL %~ Map.adjust (spsDelegatorsL %~ Set.insert stakeCred) stakePool)
+          (\accountState -> certPStateL %~ unDelegReDelegStakePool stakeCred accountState (Just stakePool))
+          mAccountState
     delegVote dRep cState =
       let cState' =
-            processDRepUnDelegation stakeCred mCurDelegatee cState
+            cState
               & certDStateL . accountsL
                 %~ adjustAccountState (dRepDelegationAccountStateL ?~ dRep) stakeCred
-          dReps
-            | preserveIncorrectDelegation = cState ^. certVStateL . vsDRepsL
-            | otherwise = cState' ^. certVStateL . vsDRepsL
-       in case dRep of
-            DRepCredential targetDRep
-              | Just dRepState <- Map.lookup targetDRep dReps ->
-                  let dRepState' = dRepState {drepDelegs = Set.insert stakeCred (drepDelegs dRepState)}
-                   in cState' & certVStateL . vsDRepsL .~ Map.insert targetDRep dRepState' dReps
-            _ -> cState'
-
-processDRepUnDelegation ::
-  ConwayEraCertState era =>
-  Credential 'Staking ->
-  Maybe Delegatee ->
-  CertState era ->
-  CertState era
-processDRepUnDelegation _ Nothing cState = cState
-processDRepUnDelegation stakeCred (Just delegatee) cState =
-  case delegatee of
-    DelegStake _ -> cState
-    DelegVote dRep -> cState & certVStateL .~ unDelegVote (cState ^. certVStateL) dRep
-    DelegStakeVote _sPool dRep -> cState & certVStateL .~ unDelegVote (cState ^. certVStateL) dRep
-  where
-    unDelegVote vState = \case
+              & maybe
+                (certVStateL %~ insertDRepDeleg dRep)
+                (\accountState -> certVStateL %~ unDelegReDelegDRep stakeCred accountState (Just dRep))
+                mAccountState
+       in if preserveIncorrectDelegation
+            then
+              cState
+                & certDStateL . accountsL %~ adjustAccountState (dRepDelegationAccountStateL ?~ dRep) stakeCred
+                & certVStateL %~ insertDRepDeleg dRep
+            else cState'
```

Comment:
Let's get rid of this tick. We can do it now, since `cState'` with these changes is only used once.
```suggestion
            else 
              cState
                & certDStateL . accountsL
                  %~ adjustAccountState (dRepDelegationAccountStateL ?~ dRep) stakeCred
                & maybe
                  (certVStateL %~ insertDRepDeleg dRep)
                  (\accountState -> certVStateL %~ unDelegReDelegDRep stakeCred accountState (Just dRep))
                  mAccountState
```